components/eventing-controller/pkg/ems/api/events/types/grant_type.go

```diff
@@ -3,3 +3,11 @@ package types
 type GrantType string
 
 const GrantTypeClientCredentials GrantType = "client_credentials"
+
+func IsInvalidGrantType(value string) bool {
+	return value != string(GrantTypeClientCredentials)
+}
+
+func GetGrantType(_ string) GrantType {
+	return GrantTypeClientCredentials
+}
```

Comment:
Why do we need this function? We can use the already exported const `GrantTypeClientCredentials`.

Reply:
The idea was to keep it extensible for future. If there are more values tomorrow, we pass the string value as a parameter and then return the `GrantType` value